pkg/kubelet/cm/devicemanager/topology.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package devicemanager
+
+import (
+	"reflect"
+
+	"k8s.io/klog"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/socketmask"
+
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+)
+
+type topology struct {
+	largestSocket int64
```

Comment:
Having a device specific attributes like `largestSocket` and `deviceMask` in a generic topology struct in Device Manager seems problematic a bit. This makes the whole design read more like "NUMA Topology awareness in node" instead of "Topology awareness in node", which does not quite align with description in https://github.com/kubernetes/enhancements/issues/693 unless I missed sth. /cc @vishh 

Per WG discussion we believe we are very glad to see topology awareness in node start from NUMA while I just expect a more generic struct and implementation which could apply to NUMA, GPU and other up-coming  topology aware devices etc.  /cc @RenaudWasTaken